### DISCUSSION TITLE
Multiple loggers from different instances of the same application writing to the same file

### DISCUSSION THREAD

**Discussion by varkey98** [gabime/spdlog#3178] [Q&A] [answered]
We have a library which uses spdlog. Inside the library, we only have a single logger writing to the file. Now there can be multiple instances of our library loaded into the memory by the application, all of them would be writing logs to the same file. How to program it? 
This is what we do now, which results in logs from a single instance being written to the file:
```
  auto sink = std::make_shared<spdlog::sinks::rotating_file_sink_mt>(
      log_file, max_file_size, max_files);
  spdlog::init_thread_pool(spdlog::details::default_async_q_size, 1U);
  auto tp = spdlog::thread_pool();
  logger = std::make_shared<spdlog::async_logger>(
      service_name, sink.value(), tp,
      spdlog::async_overflow_policy::overrun_oldest);

  spdlog::initialize_logger(logger);
  logger->set_pattern("[%Y-%m-%d %T.%e %z] [%^%l%$] [%n] %v");
  logger->set_level(spdlog_level);
  spdlog::flush_every(flush_timeout);
```

This is the error getting output from the other instance which is not writing the logs:
```
[*** LOG ERROR #0103 ***] [2024-09-11 15:12:15] [random] async log: thread pool doesn't exist anymore
```

**Comment by tt4g** [comment, marked as answer]
> ```
> [*** LOG ERROR #0103 ***] [2024-09-11 15:12:15] [random] async log: thread pool doesn't exist anymore
> ```

This error message reports that the thread pool used by the async logger has been deleted.
Please redesign your application so that the thread pool is not deleted.

https://github.com/gabime/spdlog/blob/2169a6f6ae5208e181be6184ed3f5121095fc280/include/spdlog/async_logger-inl.h#L33-L49

And writing to a single file from multiple processes is not safe: #937, #2951, #2633

**Reply by varkey98** [reply]
got it, thanks
